website/www/site/content/en/documentation/programming-guide.md

```diff
@@ -683,18 +687,18 @@ that for you. Your `@ProcessElement` method should accept a parameter tagged wit
 `@Element`, which will be populated with the input element. In order to output
 elements, the method can also take a parameter of type `OutputReceiver` which
 provides a method for emitting elements. The parameter types must match the input
-and output types of your `DoFn` or the framework will raise an error. Note: @Element and
-OutputReceiver were introduced in Beam 2.5.0; if using an earlier release of Beam, a
-ProcessContext parameter should be used instead.
+and output types of your `DoFn` or the framework will raise an error. Note: `@Element` and
+`OutputReceiver` were introduced in Beam 2.5.0; if using an earlier release of Beam, a
+`ProcessContext` parameter should be used instead.
 {{< /paragraph >}}
 
 {{< paragraph class="language-py" >}}
 Inside your `DoFn` subclass, you'll write a method `process` where you provide
 the actual processing logic. You don't need to manually extract the elements
 from the input collection; the Beam SDKs handle that for you. Your `process`
-method should accept an object of type `element`. This is the input element and
-output is emitted by using `yield` or `return` statement inside `process`
-method.
+method should accept an argument `element`, which is the input element, emit
+output value(s) by using `yield` statements. You can also use a `return`
+statement if you are sure the method will return only a single output value.
```

Comment:
The reason that `yield` is used is that `process` should return an _iterable_. `yield` is the way to pass individual elements from an iterable (does that make sense?)

Return statements can be used, but they must return an iterable - for example a list. So, for example, these functions are equivalent:

```
def process(self, elm):
  yield elm
```

and
```
def process(self, elm):
  return [elm]
```

Reply:
So, perhaps I would write something like...

```
Your `process` method should accept an argument `element`, which is the input element, 
and return an iterable with its output values. You can accomplish this by emitting individual 
elements with `yield` statements. You can also use a `return` statement with an iterable,
like a list or a generator.
```

I am not married to the phrasing - whichever phrasing you choose is fine. But I think it's important to mention the fact that an iterable is expected.

Thanks @epicfaace

Reply:
That sounds good!